### PR TITLE
Fixes #33996 - Clean up deprecated methods

### PR DESCRIPTION
**Pull request by tbrisker**
Cleans up all methods that have been marked for removal in 3.2 or older.

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -159,19 +159,6 @@ def auto_complete_controller_name
     controller.respond_to?(:auto_complete_controller_name) ? controller.auto_complete_controller_name : controller_name
   end
 
-  def auto_complete_search(name, val, options = {})
```
Comment (ezr-ondrej) [inline app/helpers/application_helper.rb:162]: This is still used in https://github.com/theforeman/foreman_remote_execution/blob/972e8536f602068445527b3813609fa543694851/app/views/job_templates/_custom_tabs.html.erb#L4
I'll try to do something about that before merging :)

**Comment by tbrisker** [comment]
opened https://github.com/theforeman/foreman_salt/pull/157 and https://github.com/theforeman/foreman_openscap/pull/515 to clean up the remaining `find_or_create` calls

**Comment by ezr-ondrej** [comment]
I guess we have still tests for those methods, that needs to also be deleted.
HostJail has test for puppetmaster and the JS tests have the taxonomies selectors and changers tests it seems

File: app/registries/foreman/plugin.rb [hunk 2]
```diff
@@ -77,11 +77,6 @@ def global_registry(name, registry)
         end
       end
 
-      def medium_providers
```
Comment (tbrisker) [inline app/registries/foreman/plugin.rb:80]: Looks like katello is still using this one...

Comment (tbrisker) [reply app/registries/foreman/plugin.rb:80]: https://github.com/Katello/katello/pull/9813

Comment (tbrisker) [reply app/registries/foreman/plugin.rb:80]: looks like this deprecation was added to the wrong place, the instance method wasn't deprecated and replaced with the new name. Updated it, but we'll need to either merge the katello pr before this (and break katello) to get katello tests green here or merge this pr (and break katello) before getting tests green on the katello PR. or merge both at the same time and hope nothing else breaks.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (ezr-ondrej) [reply app/helpers/application_helper.rb:162]: https://github.com/theforeman/foreman_remote_execution/pull/672

File: app/registries/foreman/plugin.rb (hunk 2 above)
Comment (ezr-ondrej) [reply app/registries/foreman/plugin.rb:80]: Shall we then have a separate PR just for a better visibility?

Isn't the instance method just calling the class method?

Comment (tbrisker) [reply app/registries/foreman/plugin.rb:80]: 🤔 that could make sense and allow merging the katello fix before this one, https://projects.theforeman.org/issues/34002

**Comment by tbrisker** [comment]
[test katello]

File: app/models/host/base.rb [hunk 3]
```diff
@@ -138,11 +138,6 @@ def self.import_host(hostname, certname = nil)
       host
     end
 
-    def import_facts(facts, source_proxy = nil)
```
Comment (tbrisker) [inline app/models/host/base.rb]: Looks like despite being deprecated for a while, this method is still used in many places 😞

**Review by ezr-ondrej — APPROVED** [review]
I'd love if we could open a PR for the `import_facts` method, so we can start cleaning up the use cases, but if we don't this is still nice clean up.
Thanks @tbrisker ! :+1:

**Comment by tbrisker** [comment]
https://github.com/theforeman/foreman/pull/8961

**Pull request merged**